source/server/listener_impl.cc

```diff
@@ -378,8 +377,12 @@ ListenerImpl::ListenerImpl(const envoy::config::listener::v3::Listener& config,
               cx_limit_runtime_key_, config_.name());
   }
 
+  filter_chain_manager_ = std::make_unique<FilterChainManagerImpl>(
+      addresses_, listener_factory_context_->parentFactoryContext(), initManager()),
```

Comment:
It is only for the parameter `addresses_`, since the addresses are filled up late, then I move the creation of filter chain manager late. Actually, the filter chain manager only uses the addresses for the log.